lib/codegen/CreateTerm.cpp

```diff
@@ -1277,6 +1277,11 @@ bool make_function(
     call->setTailCallKind(llvm::CallInst::TCK_MustTail);
     retval = call;
   } else {
+    bool is_apply_rule = name.substr(0, 11) == "apply_rule_";
```

Comment:
I see that the function signature has a `bool apply`. Is it possible this is already what you require?

Reply:
I have noticed that but I am not sure what this parameter is supposed to do. I checked all call sites for `make_function` they all pass `true` to this parameter, including one call site from `make_side_condition_function`: https://github.com/runtimeverification/llvm-backend/blob/3b3c19d752797e34fe4a8199bbe76d84f2cf8c41/lib/codegen/CreateTerm.cpp#L1429-L1431

Unless I am missing some call site, it seems that this parameter is dead code.

Reply:
OK I was mixing things. You are correct that we can use the `apply` parameter. I have pushed a commit that does. The first of the boolean parameters, i.e. `tailcc`, is the one that is always passed `true`.